### PR TITLE
Makes the min number of discoverable tablets with low rep lag configurable.

### PR DESCRIPTION
**Pull request by mpawliszyn**
I signed it!

### CODE REVIEW

**Comment by sougou** [comment]
This is good work. If I understand correctly, this works better than the previous algorithm because of the added sort.

Can you add some unit tests? Since this is on the serving path, we need full code coverage: essentially, a test that shows that the sorting works as intended. You also have permission to delete tests that may have become obsolete.

Coverage command: `go test -coverprofile=/tmp/c.out; go tool cover -html=/tmp/c.out`

**Comment by mpawliszyn** [comment]
For sure @sougou I'll add some testing. I think the previous algorithm was basically a one pass find the min so not technically better... I needed the sort since I'm looking for potentially more than 1 tablet. I am assuming that the set of tablets is not too big.

**Comment by mpawliszyn** [comment]
@sougou So I'm running into trouble actually getting a sort to happen when we only need a minimum of 2 tablets. The other mean code will remove them slowly enough that the sort never matters.

Anyone know of a good way to modify flags for tests? Without making too much public?

**Comment by mpawliszyn** [comment]
Not sure about my testing but it is the best I could come up with for now. Comments are welcome.

File: go/vt/discovery/replicationlag.go [hunk 1]
```diff
@@ -19,35 +19,47 @@ package discovery
...
 var (
 	// lowReplicationLag defines the duration that replication lag is low enough that the VTTablet is considered healthy.
 	lowReplicationLag            = flag.Duration("discovery_low_replication_lag", 30*time.Second, "the replication lag that is considered low enough to be healthy")
-	highReplicationLagMinServing = flag.Duration("discovery_high_replication_lag_minimum_serving", 2*time.Hour, "the replication lag that is considered too high when selecting miminum 2 vttablets for serving")
+	highReplicationLagMinServing = flag.Duration("discovery_high_replication_lag_minimum_serving", 2*time.Hour, "the replication lag that is considered too high when selecting the minimum num vttablets for serving")
+	minNumTablets                = flag.Int("min_number_serving_vttablets", 2, "the minimum number of vttablets that will be continue to be used even with low replication lag")
 )
 
+// TestSetMinNumTablets is a test helper function, if this is used by a production code path, something is wrong.
+func TestSetMinNumTablets(newMin int) {
```
Comment (sougou) [inline go/vt/discovery/replicationlag.go]: This can be moved to replicationlag_test.go, and renamed to testSetMin...

Comment (mpawliszyn) [reply go/vt/discovery/replicationlag.go]: Wow that works, updated. Thanks @sougou!

**Comment by sougou** [comment]
LGTM

**Pull request merged**